services/app-api/utils/auth/authorization.ts

```diff
@@ -64,7 +64,8 @@ export const isAuthorized = async (event: APIGatewayProxyEvent) => {
     try {
       payload = await verifier.verify(event.headers["x-api-key"]);
     } catch {
-      console.log("Token not valid!"); // eslint-disable-line
+      // verification failed - unauthorized
+      return false;
```

Comment:
Question: I have no problem with just swallowing an error here, but per line 72, the only output of this is `!!payload`, which I'm assuming would be falsy if there's an error during the try block. So do we even need the try/catch here and if not, should we just check for headers, and if there are headers, set payload? In fact, what will `verifier.verify` do if we pass it something undefined? Maybe I'm not seeing the big picture, but could this whole thing be reduced to just like:

```js
const isAuthed = await verifier.verify(event?.headers?.["x-api-key"]);
return !!isAuthed
```

Reply:
If we aren't gonna do anything with the error, we could just remove line 68 and let line 72 handle the return false and just add to line 67 a note that we're consciously swallowing any errors that happen here as they will bubble up a not authorized fallback response.

Reply:
Yep so I tested this without the catch and if you send an invalid token (in this case it was just old, not even a bad format) it throws an internal error, which returns a `502 Bad Gateway` to the user. If we catch it, it just returns `403 Unauthorized` which is more what I would expect. I did clarify the variable names, but left the try/catch logic